lib/targets/truffle/generation.ts

```diff
@@ -76,9 +76,20 @@ export interface ${c.name}Instance extends Truffle.ContractInstance {
 
 function generateFunction(fn: ConstantFunctionDeclaration | FunctionDeclaration): string {
   return `
-  ${fn.name}(${generateInputTypes(
+  ${fn.name}: {
+    (${generateInputTypes(
+      fn.inputs,
+    )} txDetails?: Truffle.TransactionDetails): Promise<Truffle.TransactionResponse>;
+  call(${generateInputTypes(
```

Comment:
the original issue mentioned that it `call` should be missing on state changing methods which practically is true but in theory you can still `call` them. Probably `sendTransaction` should be missing on pure and view functions as well 🤷‍♂ What do you think?

Reply:
I think `call` is valid to simulate a state changing method with call and the expected behavior in the issue is not quite right.

`sendTransaction` is already missing on constant/ view/ pure functions as I didn't modify `generateConstantFunction`

Reply:
Ok I agree 👍